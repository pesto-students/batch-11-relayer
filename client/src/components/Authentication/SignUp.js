import React, { createRef } from 'react';
import { Container } from 'reactstrap';
import SignupForm from './SignupForm';
import prepareRequest from '../../utils/requestEPLib';
import request from '../../utils/requestLib';
import '../../assets/styles/components/_signup.module.scss';

const SignUp = () => {
  const signUp = async (event, email, pass) => {
    event.preventDefault();
    console.log(email.current.value, pass.current.value);
    const requestObj = prepareRequest('signUp');
    requestObj.body = { email: email.current.value, password: pass.current.value };
    const response = await request(requestObj);
  };

  return (
    <Container>
      <h1 className="mt-5 mb-5"> Sign Up </h1>
      <SignupForm signUp={signUp} />
    </Container>
  );
};

export default SignUp;
