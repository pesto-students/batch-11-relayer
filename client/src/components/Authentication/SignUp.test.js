import React from 'react';
import { shallow } from 'enzyme';
import SignUp from './SignUp';

describe('<SignUp />', () => {
  it('renders the SignUp component', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper).toMatchSnapshot();
  });
});
