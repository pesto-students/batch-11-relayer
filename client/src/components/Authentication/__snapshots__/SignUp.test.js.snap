// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserActions /> renders the UserActions component 1`] = `ShallowWrapper {}`;
